src/sss_client/autofs/sss_autofs.c

```diff
@@ -33,6 +33,32 @@
 /* How many entries shall _sss_getautomntent_r retrieve at once */
 #define GETAUTOMNTENT_MAX_ENTRIES   512
 
+unsigned int _protocol = 0;
```

Comment:
Hi,

I think `_protocol` should be:
1) `static` at the very least.
2) strictly speaking access to this var should be thread safe.
I would prefer C11 `atomic_int` but `sss_client` in general uses `libpthread`.

Reply:
atomics are better in that they do not incur the heavy weight of acquiring a mutx (which is done via atomics anyway). So I see no problem in using that.

Reply:
And how about `thread_local`, do you see any problem there?

Reply:
Imo, this won't hurt but I'm not sure if it makes much sense.

`_Thread_local` here would be to cover case "application wants to use sssd autofs lib with different protocol version from different threads".

Frankly I can't imagine sane reason to do so. But even if there are, and we want to support this, then we should document this somehow/somewhere. Currently looking at the API itself one can't judge if there is thread specific context.

Reply:
There is no functional reason - autofs will always try to used its latest protocol version, atomic should be enough to do that.